<?xml version="1.0"?>
<launch>
    <arg name="respawn" default="false" />
    <arg name="debug" default="false" />
    <arg name="node_name" default="lucid_camera_node" />
    <arg name="config_file" default="$(find arena_camera)/config/default.yaml" />

    <arg unless="$(arg debug)" name="launch_prefix" value="" />
    <arg     if="$(arg debug)" name="launch_prefix" value="gdb -ex run --args" />

    <node name="$(arg node_name)" pkg="arena_camera" type="arena_camera_node" output="screen"
          respawn="$(arg respawn)" launch-prefix="$(arg launch_prefix)">
      <remap from="$(arg node_name)/image_raw" to="/hardware_interface/camera/image_raw" />
      <remap from="$(arg node_name)/camera_info" to="/hardware_interface/camera/camera_info" />
      <remap from="$(arg node_name)/discovery" to="/hardware_interface/driver_discovery" />
      <param name="brightness" type="int" value="70"/>
      <param name="frame_rate" type="double" value="10.0"/>
      <param name="image_encoding" type="string" value="bayer_rggb8"/>
      <param name="sensor_binning" type="bool" value="true"/>
      <param name="binning_x" type="int" value="1"/>
      <param name="binning_y" type="int" value="1"/>
      <param name="device_user_id" type="string" value="Front"/>
    </node>
</launch>
